MoreLinq/NestedLoops.cs

```diff
@@ -41,21 +41,18 @@ public static IEnumerable<Action> NestedLoops(this Action action, IEnumerable<in
             if (action == null) throw new ArgumentNullException(nameof(action));
             if (loopCounts == null) throw new ArgumentNullException(nameof(loopCounts));
 
-            using (var iter = loopCounts.GetEnumerator())
+            return _(); IEnumerable<Action> _()
             {
-                var loopCount = NextLoopCount(iter);
-                if (loopCount == null)
-                    return Enumerable.Empty<Action>(); // null loop
-                var loop = Enumerable.Repeat(action, loopCount.Value);
-                while ((loopCount = NextLoopCount(iter)) != null)
-                    loop = loop.Repeat(loopCount.Value);
-                return loop;
-            }
+                var count = loopCounts.Assert(n => n >= 0,
+                                              n => new InvalidOperationException("All loop counts must be greater than or equal to zero."))
```

Comment:
Change exception message from plural to:

> All loop counts must be greater than or equal to zero.

to singular form since it concerns one _n_:

> Invalid loop count (must be greater than or equal to zero).

Reply:
done. building running.